lib/DXIL/DxilShaderModel.cpp

```diff
@@ -494,6 +494,18 @@ const llvm::StringRef ShaderModel::FullNameFromKind(DXIL::ShaderKind sk) {
   }
 }
 
+bool ShaderModel::AllowDerivatives(DXIL::ShaderKind sk) {
+  switch (sk) {
+  case DXIL::ShaderKind::Pixel:
+  case DXIL::ShaderKind::Library:
+  case DXIL::ShaderKind::Compute:
+  case DXIL::ShaderKind::Amplification:
+  case DXIL::ShaderKind::Mesh:
+    return true;
```

Comment:
Missing `DXIL::ShaderKind::Node`, right?

Reply:
Added node.
Should all the ray tracing shader kind be added here as well?

Reply:
No, derivative ops are not allowed in DXR shaders.